specs/GetFavicon.md

```diff
@@ -0,0 +1,177 @@
+
+# Description
+We propose a new Webview2 event which would allow developers to access the current Favicon of a page and be notified when the favicon changes. This means when a page first loads, it would raise the FaviconChanged event since before parsing the HTML document there is no known favicon. DOM or JavaScript may change the Favicon, causing the event to be raised again for the same document.
+
+# Examples
+## Win32 C++ Registering a listener for favicon changes
+```cpp
+Gdiplus::GdiplusStartupInput gdiplusStartupInput;
+
+// Initialize GDI+.
+Gdiplus::GdiplusStartup(&gdiplusToken_, &gdiplusStartupInput, NULL);
+CHECK_FAILURE(m_webView2->add_FaviconChanged(
+    Callback<ICoreWebView2FaviconChangedEventHandler>(
+        [this](ICoreWebView2* sender, IUnknown* args) -> HRESULT {
+            wil::unique_cotaskmem_string url;
+            Microsoft::WRL::ComPtr<ICoreWebView2>
+                webview2;
+            CHECK_FAILURE(
+                sender->QueryInterface(IID_PPV_ARGS(&webview2)));
+
+            CHECK_FAILURE(webview2->get_FaviconUri(&url));
+            std::wstring strUrl(url.get());
+
+            webview2->GetFavicon(
+                COREWEBVIEW2_FAVICON_IMAGE_FORMAT_PNG,
+                Callback<ICoreWebView2GetFaviconCompletedHandler>(
+                    [this, strUrl](HRESULT errorCode, IStream* iconStream) -> HRESULT
+                    {
+                        CHECK_FAILURE(errorCode);
+                        Gdiplus::Bitmap iconBitmap(iconStream);
+                        wil::unique_hicon icon;
+                        if (iconBitmap.GetHICON(&icon) == Gdiplus::Status::Ok)
+                        {
+                            m_favicon = std::move(icon);
+                            SendMessage(
+                                m_appWindow->GetMainWindow(), WM_SETICON,
+                                ICON_SMALL, (LPARAM)m_favicon.get());
+                        }
+                        else
+                        {
+                            SendMessage(
+                                m_appWindow->GetMainWindow(), WM_SETICON,
+                                ICON_SMALL, (LPARAM)IDC_NO);
+                        }
+
+                        return S_OK;
+                    })
+                    .Get());
+
+            return S_OK;
+        }).Get(), &m_faviconChangedToken));
+}
+```
+## .NET / WinRT Registering a listener for favicon changes
+```c#
+webView.CoreWebView2.FaviconChanged += (object sender, Object arg) =>
+{
+    string value = webView.CoreWebView2.FaviconUri;
+    System.IO.Stream stream = await webView.CoreWebView2.GetFaviconAsync(
+      CoreWebView2FaviconImageFormat.Png);
+    if (stream == null || stream.Length == 0)
+        this.Icon = null;
+    else
+        this.Icon = BitmapFrame.Create(stream);
+};
+```
+# API Notes
+Note that even if a web page does not have a Favicon, the FaviconChanged event
+is raised when the page is navigated to. The Favicon would be an
```

Comment:
I think, from discussion, this statement is incorrect. If the previous page had no favicon, then you don't get a FaviconChanged event on the new page, since there is still no favicon.

Reply:
The assumption was that there was a fav icon loaded before. Updated to remove that ambiguity.